### ISSUE TITLE
Missing 'Check for Updates' and 'Show Log Files' after running a Shiny app

### ISSUE THREAD

**Issue by larspijnappel** [rstudio/rstudio#4852]
(As requested by Kevin Ushey)

When starting my latest RStudio version (1.2.1335), I can choose 'Check for Updates' from the Help menu and all is working fine. However, after starting and closing a Shiny application, this option is gone from the Help menu. Only restarting the IDE solves this issue again.

![image](https://user-images.githubusercontent.com/12656725/58428757-d7335180-80a3-11e9-94ad-f3f5401ce967.png)

How to reproduce:

1. Create New File > Shiny Web app ...
2. Check Help > Check for Updates << option is available
3. Click Run App, then close it
4. Check Help > Check for Updates << option is gone

![image](https://user-images.githubusercontent.com/12656725/58428785-e914f480-80a3-11e9-81f9-071102bc1a12.png)

**Comment by pascaloettli** [comment]
Please provide the system details, as requested when opening a new issue

**Comment by riccardoporreca** [comment]
Same behavior also when knitting a document.

How to reproduce:
1. Open RStudio, "Check for Updates" appears in the "Help" menu
2. Open a .Rmd document
3. Knit (Ctrl+Shift+K)
4. "Check for Updates" not in the "Help" menu
5. Close and re-open RStudio, "Check for Updates" is back

```
RStudio
Version 1.2.1335
© 2009-2019 RStudio, Inc.
Build 4 (f1ac3452)
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.1 Chrome/69.0.3497.128 Safari/537.36
```

**Comment by engineerchange** [comment]
+1 - just noticed the same behavior as @riccardoporreca describes while knitting.

```
RStudio
Version 1.2.5042
@ 2009-2020 RStudio, Inc.
"Double Marigold" (e4a1c219, 2020-04-01)
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.1 Chrome/69.0.3497.128 Safari/537.36
```

**Comment by gtritchie** [comment]
There are other commands missing as well:

Help / Diagnostics / Show Log Files
Help / Diagnostics / Write Diagnostics Report

<img width="279" alt="screenshot of Help / Diagnostics submenu highlighting missing commands" src="https://user-images.githubusercontent.com/10569626/87814434-97d04300-c818-11ea-87b5-fad5da761edf.png">

These are _supposed_ to be make visible here (suggesting this code is not running?).

https://github.com/rstudio/rstudio/blob/a88a74b6780e8bebaea1e44adf555ee9c8750d84/src/gwt/src/org/rstudio/studio/client/application/ui/impl/DesktopApplicationHeader.java#L122-L125

**Comment by jmcphers** [comment]
This is a systemic problem which will require a careful fix. 

The root of the issue is that we don't have distinct per-window command states. Consequently, when a satellite window (hosting a Shiny app, etc.) initializes its command system, a number of commands get set to their initial states as defined in `Commands.cmd.xml`. *Check for Updates* is hidden here, for example:

https://github.com/rstudio/rstudio/blob/ef42351960128ddd6e910a315016529c6dae0292/src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml#L2956-L2959

All of the MRU commands have this problem, too, which is the root of the problem in https://github.com/rstudio/rstudio/issues/7126 and probably others.

https://github.com/rstudio/rstudio/blob/ef42351960128ddd6e910a315016529c6dae0292/src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml#L1198-L1212

When the command system initializes in the satellite window, it dutifully pushes all of the commands' state into the desktop frame (this bug does not exist on RStudio Server), meaning that commands that were visible are now hidden again. These commands normally get re-enabled in the main window, but the machinery to re-enable them doesn't get invoked in satellites, so they stay hidden.

**Comment by R-Hannibal** [comment]
Verified Fixed on: RStudio Version - 1.4.703, R Version - 4.0.2, Windows 10 x64 - Desktop 

Comparison testing between current daily and current stable 1.3.1073, menu entries now remain consistent and accessible in the IDE after opening and running a Shiny Web App file.
